python/lsst/afw/image/_visitInfo.py

```diff
@@ -22,12 +22,105 @@
 """
 
 from lsst.utils.deprecated import deprecate_pybind11
+from lsst.utils import continueClass
 from ._imageLib import VisitInfo
 
 
 __all__ = []
 
 
+@continueClass
+class VisitInfo:  # noqa: F811
+    def copyWith(
```

Comment:
General comment: this doesn't quite seem to cover all the items. I think `[boresightHourAngle, boresightParAngle, localEra]` are missing. And then there's `isPersistable`, but I wonder where that ever comes from and if it makes sense to set... That feels like a question for @timj perhaps?

Reply:
Oh, I guess I did miss those. I'll fix and update the PR.

Reply:
Those must all be derived quantities. I don't see any of them in the constructor for `VisitInfo`: https://github.com/lsst/afw/blob/main/include/lsst/afw/image/VisitInfo.h#L100-L130

Reply:
Hmmm, I wonder how they get in there then, given that the object is immutable. Presumably as part of the init, so these will show up as intended after all?

Reply:
They aren't stored anywhere. Each time they are called they are calculated in C++:
https://github.com/lsst/afw/blob/main/src/image/VisitInfo.cc#L633
https://github.com/lsst/afw/blob/main/src/image/VisitInfo.cc#L635
https://github.com/lsst/afw/blob/main/src/image/VisitInfo.cc#L637-L649

They just look like attributes because they are wrapped as properties by pybind11: https://github.com/lsst/afw/blob/main/python/lsst/afw/image/_visitInfo.cc

Reply:
Ah, interesting, thanks!